src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,19 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+
     private final int[] decodeInts(final String string) {
-        final int nValues = ParsingUtils.split(string, INT_DECODE_ARRAY, ',');
+        // find number of CSV items to set array size
+        int numItems = string.split(",").length;
+        String[] intDecodeArray = new String[numItems];
+        final int nValues = ParsingUtils.split(string, intDecodeArray, ',');
+        if ( nValues != numItems ){
```

Comment:
This is defensive programming, or in this case paranoid programming. It will be removed.